modules/twinkleblock.js

```diff
@@ -0,0 +1,879 @@
+//<nowiki>
+
+
+(function($){
+
+var api = new mw.Api(), result, relevantUser = mw.config.get('wgRelevantUserName'), relevantUserName, formCleared = false;
+
+/*
+ ****************************************
+ *** twinkleblock.js: Block module
+ ****************************************
+ * Mode of invocation:     Tab ("Block")
+ * Active on:              any page with relevant user name (userspace, contribs, etc.)
+ * Config directives in:   [soon to be TwinkleConfig]
+ */
+
+Twinkle.block = function twinkleblock() {
+  // should show on Contributions pages, anywhere there's a relevant user
+  if ( Morebits.userIsInGroup('sysop') && relevantUser ) {
+    Twinkle.addPortletLink(Twinkle.block.callback, "Block", "tw-block", "Block relevant user" );
+  }
+};
+
+// Contains the current block info in an object
+// Once filled, it will look something like:
+// { expiry: <some date>, anononly: true, ... }
+Twinkle.block.currentBlockInfo = undefined;
+
+Twinkle.block.callback = function twinkleblockCallback() {
+
+  var Window = new Morebits.simpleWindow( 650, 530 );
+  // need to be verbose about who we're blocking
+  Window.setTitle( "Block or template " + relevantUser );
```

Comment:
I don't like using "template" as a verb, as it's not very clear. I'd prefer a simple title like "Block user or issue a block template"